File: server/datasets/java_source-code/Multiple-G1/renamed_45_bst.java
class Node_9 {
    int val_9;
    Node_9 lhs_9, rhs_9;
    public Node_9(int item) {
        val_9 = item;
        lhs_9 = rhs_9 = null;
    }
}
public class Tree_9 {
    public static void main(String[] args) {
        Tree_9 bst = new Tree_9();
        bst.ins_9(50);
        bst.srch_9(50);
    }
    Node_9 rt_9;
    public void ins_9(int val_9) {
        rt_9 = in_rec_9(rt_9, val_9);
    }
    Node_9 in_rec_9(Node_9 rt_9, int val_9) {
        if (rt_9 == null) return new Node_9(val_9);
        if (val_9 < rt_9.val_9) rt_9.lhs_9 = in_rec_9(rt_9.lhs_9, val_9);
        else if (val_9 > rt_9.val_9) rt_9.rhs_9 = in_rec_9(rt_9.rhs_9, val_9);
        return rt_9;
    }
    Node_9 srch_9(int val_9) {
        return sr_rec_9(rt_9, val_9);
    }
    Node_9 sr_rec_9(Node_9 rt_9, int val_9) {
        if (rt_9 == null || rt_9.val_9 == val_9) return rt_9;
        if (rt_9.val_9 < val_9) return sr_rec_9(rt_9.rhs_9, val_9);
        return sr_rec_9(rt_9.lhs_9, val_9);
    }
}

File: server/datasets/controlled_benchmarks/python/multiple_files/multiple_g1/renamed_45_bst.py
class Node_9:
    def __init__(self, val_9):
        self.val_9 = val_9
        self.lhs_9 = None
        self.rhs_9 = None

class Tree_9:
    def __init__(self):
        self.rt_9 = None

    def ins_9(self, val_9):
        if self.rt_9 is None:
            self.rt_9 = Node_9(val_9)
        else:
            self._ins_9_rec(self.rt_9, val_9)

    def _ins_9_rec(self, nd_9, val_9):
        if val_9 < nd_9.val_9:
            if nd_9.lhs_9 is None:
                nd_9.lhs_9 = Node_9(val_9)
            else:
                self._ins_9_rec(nd_9.lhs_9, val_9)
        elif val_9 > nd_9.val_9:
            if nd_9.rhs_9 is None:
                nd_9.rhs_9 = Node_9(val_9)
            else:
                self._ins_9_rec(nd_9.rhs_9, val_9)

    def srch_9(self, val_9):
        return self._srch_9_rec(self.rt_9, val_9)

    def _srch_9_rec(self, nd_9, val_9):
        if nd_9 is None or nd_9.val_9 == val_9:
            return nd_9
        if val_9 < nd_9.val_9:
            return self._srch_9_rec(nd_9.lhs_9, val_9)
        return self._srch_9_rec(nd_9.rhs_9, val_9)
